dataplane/handlers/interface.go

```diff
@@ -213,61 +213,102 @@ func (ni *Interface) reconcile(config *oc.Interface) {
 			log.V(1).Infof("setting interface %s hw-addr %q", tapName, config.GetEthernet().GetMacAddress())
 			if err := kernel.SetInterfaceHWAddr(config.GetName(), config.GetEthernet().GetMacAddress()); err != nil {
 				log.Warningf("Failed to set mac address of port: %v", err)
-				return
+			}
+		}
+	} else {
+		if state.GetEthernet().GetHwMacAddress() != state.GetEthernet().GetMacAddress() {
+			log.V(1).Infof("resetting interface %s hw-addr %q", tapName, state.GetEthernet().GetHwMacAddress())
+			if err := kernel.SetInterfaceHWAddr(config.GetName(), state.GetEthernet().GetHwMacAddress()); err != nil {
+				log.Warningf("Failed to set mac address of port: %v", err)
 			}
 		}
```

Comment:
Add a comment on why this gets reset (perhaps link to https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-interfaces.html#interfaces-interface-ethernet-state-mac-address)

Reply:
done